From e79a9500cc2ff58e02af1e115efebda025c5480a Mon Sep 17 00:00:00 2001
From: Ryan Brue <ryanbrue.dev@gmail.com>
Date: Sun, 29 Sep 2024 16:04:19 -0500
Subject: [PATCH] fix: use /usr/libexec/.. for plugins instead of /usr/lib/..

This commit changes plugins to use /usr/libexec instead of /usr/lib. Since the plugins contain executables, the more apt directory for this need is /usr/libexec. Fedora is requiring this in order to package pop-launcher and cosmic-launcher, so this patch will be used for downstream Fedora, but if upstream would like to adopt this change, I've brought it here just in case! :)

Signed-off-by: Ryan Brue <ryanbrue.dev@gmail.com>
---
 README.md                                  |  4 ++--
 debian/pop-launcher-system76-power.install |  2 +-
 debian/pop-launcher.install                |  4 ++--
 debian/pop-launcher.links                  | 22 +++++++++++-----------
 debian/rules                               |  2 +-
 plugins/src/scripts/mod.rs                 |  2 +-
 src/lib.rs                                 |  2 +-
 7 files changed, 19 insertions(+), 19 deletions(-)

diff --git a/README.md b/README.md
index 54fefcb..c67f9c0 100644
--- a/README.md
+++ b/README.md
@@ -45,7 +45,7 @@ just plugins="calc desktop_entries files find pop_shell pulse recent scripts ter
 
 - User-local plugins: `~/.local/share/pop-launcher/plugins/{plugin}/`
 - System-wide install for system administrators: `/etc/pop-launcher/plugins/{plugin}/`
-- Distribution packaging: `/usr/lib/pop-launcher/plugins/{plugin}/`
+- Distribution packaging: `/usr/libexec/pop-launcher/plugins/{plugin}/`
 
 ## Plugin Config
 
@@ -79,7 +79,7 @@ A plugin's metadata is defined `pop-launcher/plugins/{plugin}/plugin.ron`.
 
 - User-local scripts: `~/.local/share/pop-launcher/scripts`
 - System-wide install for system administrators: `/etc/pop-launcher/scripts`
-- Distribution packaging: `/usr/lib/pop-launcher/scripts`
+- Distribution packaging: `/usr/libexec/pop-launcher/scripts`
 
 Example script
 <details>
diff --git a/debian/pop-launcher-system76-power.install b/debian/pop-launcher-system76-power.install
index 72e3852..698c3b2 100644
--- a/debian/pop-launcher-system76-power.install
+++ b/debian/pop-launcher-system76-power.install
@@ -1 +1 @@
-/usr/lib/pop-launcher/scripts/system76-power
\ No newline at end of file
+/usr/libexec/pop-launcher/scripts/system76-power
\ No newline at end of file
diff --git a/debian/pop-launcher.install b/debian/pop-launcher.install
index 6852f59..3c03459 100644
--- a/debian/pop-launcher.install
+++ b/debian/pop-launcher.install
@@ -1,3 +1,3 @@
 /usr/bin/
-/usr/lib/pop-launcher/plugins/
-/usr/lib/pop-launcher/scripts/session/
\ No newline at end of file
+/usr/libexec/pop-launcher/plugins/
+/usr/libexec/pop-launcher/scripts/session/
\ No newline at end of file
diff --git a/debian/pop-launcher.links b/debian/pop-launcher.links
index 99b5032..3549763 100644
--- a/debian/pop-launcher.links
+++ b/debian/pop-launcher.links
@@ -1,11 +1,11 @@
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/calc/calc
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/desktop_entries/desktop-entries
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/files/files
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/find/find
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/pop_shell/pop-shell
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/pulse/pulse
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/recent/recent
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/scripts/scripts
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/terminal/terminal
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/web/web
-/usr/bin/pop-launcher /usr/lib/pop-launcher/plugins/cosmic_toplevel/cosmic-toplevel
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/calc/calc
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/desktop_entries/desktop-entries
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/files/files
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/find/find
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/pop_shell/pop-shell
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/pulse/pulse
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/recent/recent
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/scripts/scripts
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/terminal/terminal
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/web/web
+/usr/bin/pop-launcher /usr/libexec/pop-launcher/plugins/cosmic_toplevel/cosmic-toplevel
diff --git a/debian/rules b/debian/rules
index 30cad5a..30d994b 100755
--- a/debian/rules
+++ b/debian/rules
@@ -32,4 +32,4 @@ override_dh_auto_install:
 
 override_dh_fixperms:
 	dh_fixperms
-	chmod +x debian/pop-launcher/usr/lib/pop-launcher/scripts/**/*.sh
\ No newline at end of file
+	chmod +x debian/pop-launcher/usr/libexec/pop-launcher/scripts/**/*.sh
\ No newline at end of file
diff --git a/plugins/src/scripts/mod.rs b/plugins/src/scripts/mod.rs
index b0b75d7..54faac7 100644
--- a/plugins/src/scripts/mod.rs
+++ b/plugins/src/scripts/mod.rs
@@ -15,7 +15,7 @@ use tokio::process::Command;
 
 const LOCAL_PATH: &str = ".local/share/pop-launcher/scripts";
 const SYSTEM_ADMIN_PATH: &str = "/etc/pop-launcher/scripts";
-const DISTRIBUTION_PATH: &str = "/usr/lib/pop-launcher/scripts";
+const DISTRIBUTION_PATH: &str = "/usr/libexec/pop-launcher/scripts";
 
 pub async fn main() {
     let mut requests = json_input_stream(async_stdin());
diff --git a/src/lib.rs b/src/lib.rs
index e626e9d..3cb9044 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -19,7 +19,7 @@ pub const LOCAL_PLUGINS: &str = concatcp!(LOCAL, "/plugins");
 pub const SYSTEM: &str = "/etc/pop-launcher";
 pub const SYSTEM_PLUGINS: &str = concatcp!(SYSTEM, "/plugins");
 
-pub const DISTRIBUTION: &str = "/usr/lib/pop-launcher";
+pub const DISTRIBUTION: &str = "/usr/libexec/pop-launcher";
 pub const DISTRIBUTION_PLUGINS: &str = concatcp!(DISTRIBUTION, "/plugins");
 
 pub const PLUGIN_PATHS: &[&str] = &[LOCAL_PLUGINS, SYSTEM_PLUGINS, DISTRIBUTION_PLUGINS];
